Core/Core.php

```diff
@@ -445,7 +445,6 @@ function property_exists($object_or_class, string $property): bool {}
  * @link https://secure.php.net/manual/en/function.trait-exists.php
  * @since 5.4
  */
-#[Pure(true)]
```

Comment:
It looks like this change is incorrect since `trait_exists` actually depends on global state. For example 
```
var_dump(trait_exists('MyTrait')); // false

trait MyTrait {}

var_dump(trait_exists('MyTrait')); // true
```
And at the same time this function doesn't produce any side effects, that's why it should be marked with `#[Pure(true)]` attribute.

Reply:
Per the commit message these methods all trigger the autoloader which can (and is really intended) to have side effects namely defining symbols but nothing stopping it doing other things.

Reply:
Ah, yes. I agree here. The function can produce indirect side effects via autoloader.